File: app/src/main/java/com/example/megaport/mynews/Models/Article.java
package com.example.megaport.mynews.Models;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.List;

// Models used to manipulate an article object

public class Article {

    @SerializedName("web_url")
    @Expose
    private String webUrl;
    @SerializedName("snippet")
    @Expose
    private String snippet;
    @SerializedName("pub_date")
    @Expose
    private String pubDate;
    @SerializedName("document_type")
    @Expose
    private String documentType;
    @SerializedName("new_desk")
    @Expose
    private String newDesk;
    @SerializedName("headline")
    @Expose
    private Headline headline;
    @SerializedName("type_of_material")
    @Expose
    private String typeOfMaterial;
    @SerializedName("section_name")
    @Expose
    private String sectionName;
    @SerializedName("uri")
    @Expose
    private String uri;
    @SerializedName("section")
    @Expose
    private String section;
    @SerializedName("subsection")
    @Expose
    private String subsection;
    @SerializedName("title")
    @Expose
    private String title;
    @SerializedName("url")
    @Expose
    private String url;
    @SerializedName("published_date")
    @Expose
    private String publishedDate;
    @SerializedName("multimedia")
    @Expose
    private List<Multimedium> multimedia = null;
    @SerializedName("media")
    @Expose
    private List<Medium> media = null;

    public String getSection() {
        return section;
    }

    public String getSubsection() {
        return subsection;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getUrl() {
        return url;
    }

    public String getPublishedDate() {
        return publishedDate;
    }

    public List<Multimedium> getMultimedia() {
        return multimedia;
    }

    public List<Medium> getMedia() {
        return media;
    }

    public void setSection(String section) {
        this.section = section;
    }

    public void setSubsection(String subsection) {
        this.subsection = subsection;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public void setPublishedDate(String publishedDate) {
        this.publishedDate = publishedDate;
    }

    public void setMultimedia(List<Multimedium> multimedia) {
        this.multimedia = multimedia;
    }

    public void setMedia(List<Medium> media) {
        this.media = media;
    }

    public String getWebUrl() {
        return webUrl;
    }

    public String getPubDate() {
        return pubDate;
    }

    public String getNewDesk() {
        return newDesk;
    }

    public String getSectionName() {
        return sectionName;
    }

    public String getSnippet() {
        return snippet;
    }



}

File: app/src/main/java/com/example/megaport/mynews/Models/Article.kt
package com.example.megaport.mynews.Models

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

// Models used to manipulate an article object

class Article {

    @SerializedName("web_url")
    @Expose
    val webUrl: String? = null
    @SerializedName("snippet")
    @Expose
    val snippet: String? = null
    @SerializedName("pub_date")
    @Expose
    val pubDate: String? = null
    @SerializedName("document_type")
    @Expose
    private val documentType: String? = null
    @SerializedName("new_desk")
    @Expose
    val newDesk: String? = null
    @SerializedName("headline")
    @Expose
    private val headline: Headline? = null
    @SerializedName("type_of_material")
    @Expose
    private val typeOfMaterial: String? = null
    @SerializedName("section_name")
    @Expose
    val sectionName: String? = null
    @SerializedName("uri")
    @Expose
    private val uri: String? = null
    @SerializedName("section")
    @Expose
    var section: String? = null
    @SerializedName("subsection")
    @Expose
    var subsection: String? = null
    @SerializedName("title")
    @Expose
    var title: String? = null
    @SerializedName("url")
    @Expose
    var url: String? = null
    @SerializedName("published_date")
    @Expose
    var publishedDate: String? = null
    @SerializedName("multimedia")
    @Expose
    var multimedia: List<Multimedium>? = null
    @SerializedName("media")
    @Expose
    var media: List<Medium>? = null


}
